Set Content-Type header for response to invalid bodies

andrewjli opened the pull request:
When Paperwork validates a body and then fails (i.e. the body doesn't meet the requirements), the response does not have a `Content-Type` header set. This should really be set to some variation of `application/json` and the response should be sent as proper JSON, not JSON as a preformatted string, which is currently being sent.

By changing [paperwork.js#L154](https://github.com/lperrin/paperwork/blob/master/paperwork.js#L154) to `res.json` instead of `res.end`, Express automatically sets the `Content-Type` to `application/json; charset=utf-8`. I've also updated the tests to check for usage of `json()` instead of `end()`.

For what it's worth, the [Express API guidelines](http://expressjs.com/api.html#res.end) actually say *not* to use `res.end()` if you need to send data.

lperrin commented:
Hey there,

This code was changed not to depend on express. Would it be okay to manually set the header (with res.setHeader) instead?

andrewjli commented:
Thanks for pointing that out. I had read that in the past PRs but completely forgot about it for some reason. I've updated and rebased my commits.

I also noticed that you defined the `response` object and then proceeded to create a new identical object inside `res.end`. I'm not sure why you do that, but I took liberty in changing it to use the `response` object instead.
